requestPromiseUtils.js

```diff
@@ -0,0 +1,34 @@
+const requestPromise = require('request-promise');
+const requestPromiseErrors = require('request-promise/errors');
+
+const requestWithRetries = async (requestOptions, retryHandler, computeRetryDelay) => {
+  const makeRequest = async (tryCount) => {
+    let response;
+    try {
+      response = await requestPromise(requestOptions);
+    } catch (err) {
+      if (!retryHandler || !retryHandler(err, tryCount)) {
+        throw err;
+      }
+
+      const nextTryCount = tryCount + 1;
+      const retryDelay = await ((computeRetryDelay && computeRetryDelay(err, nextTryCount)) || 1000);
```

Comment:
`await` seems unncessary since `computeRetryDelay` isn't promisy

Reply:
But it could be ... and we are in an `async` context so I thought it best to remain compatible.